### PR TITLE
chore: [IOAPPX-476] Change `StatusBar` style according to the current color scheme

### PR DESCRIPTION
**Pull request by dmnplb**
## Short description
This PR adds a `StatusBar` that changes its style according to the color scheme of the app.

## List of changes proposed in this pull request
- Add a `StatusBar` with a specific color scheme condition in the `AppStackNavigator`

| Before | After |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/a9be6e64-dc9d-4ef5-b3b2-33b48d9d509f" /> | <video src="https://github.com/user-attachments/assets/8b2a56dc-ea83-4a13-bb00-d12830aca3f8" /> | 

## How to test
Change the color scheme of your app from **Dev Settings** and randomly check the headers of the screens. Focus your attention on the screens with an existing specific `StatusBar` behavior.

### CODE REVIEW

**Review by LeleDallas — COMMENTED** [review]
In B&P I saw some issue related: 

| All payment checkout are affected  | Scan QR | IDPay  | CGN |`CGN_MERCHANTS_DISCOUNT_SCREEN`|
|--------|--------|--------|--------|--------|
| Screenshot 2025-02-03 at 18 29 48 | Screenshot 2025-02-03 at 18 27 39 | Screenshot 2025-02-03 at 18 24 39 | Screenshot 2025-02-03 at 18 20 24 | Screenshot 2025-02-03 at 18 34 21|

In these cases, I think it would be necessary to declare the `StatusBar` component again, but with the condition negated:
```
 <StatusBar  barStyle={themeType === "dark" ? "dark-content": "light-content"} />
```
This partially resolves the issue, but if the view is scrollable, the header colors may become mismatched again:

https://github.com/user-attachments/assets/ad7fa9b7-6f17-4d29-abc0-10385d8da93b

**Comment by dmnplb** [comment]
> In B&P I saw some issue related

@LeleDallas For the CGN flow, I tried to replace some logic. In a sentence: we make the custom background color persistent, even if the citizen scrolls. This change requires some changes to the `HeaderSecondLevel` component in the DS: if we set the `backgroundColor` and `variant` props, the `IconButton` will change its colors as well, because it's designed to invert the colors when dark mode kicks in. What do you think?

Here's the relative commit → [`6503bc7` (#6680)](https://github.com/pagopa/io-app/pull/6680/commits/6503bc764949ecebf38da71009a2fc6fa54e9667) and a preview of the behavior (buggy, for now):

https://github.com/user-attachments/assets/abdf9e79-b395-4abc-8fdb-8071fd759ba8

**Comment by LeleDallas** [comment]
> > In B&P I saw some issue related
> 
> @LeleDallas For the CGN flow, I tried to replace some logic. In a sentence: we make the custom background color persistent, even if the citizen scrolls. This change requires some changes to the `HeaderSecondLevel` component in the DS: if we set the `backgroundColor` and `variant` props, the `IconButton` will change its colors as well, because it's designed to invert the colors when dark mode kicks in. What do you think?
> 
> Here's the relative commit → [`6503bc7` (#6680)](https://github.com/pagopa/io-app/pull/6680/commits/6503bc764949ecebf38da71009a2fc6fa54e9667) and a preview of the behavior (buggy, for now):
> 
>  enable-discrete-transition.mp4

Sounds good!
Now I see an edge case in the video:  when the header is yellow (or a light color) I think the header text (and also the action buttons) must be dark

**Comment by LeleDallas** [comment]
⚠️In android status bar is not updating correctly⚠️

https://github.com/user-attachments/assets/77feb5a9-5127-4820-aa6b-8d3210b1e8aa

File: ts/navigation/AppStackNavigator.tsx [hunk 1]
```diff
@@ -165,6 +165,9 @@ const InnerNavigationContainer = (props: InnerNavigationContainerProps) => {
         routeNameRef.current = currentRouteName;
       }}
     >
+      <StatusBar
+        barStyle={themeType === "dark" ? "light-content" : "dark-content"}
```
[the remark this answers is not in the document]
Comment (dmnplb) [reply ts/navigation/AppStackNavigator.tsx:174]: @LeleDallas Fixed in the following commit: [`378a035` (#6680)](https://github.com/pagopa/io-app/pull/6680/commits/378a035a5ca91ecb4aa334e7f8af58e6f4835bee). Also tested on real device. It covers most cases

**Comment by dmnplb** [comment]
@LeleDallas In the last commits I fixed the wrong `StatusBar` in the main and category screens. There's still a TS bug, fixed with [this PR](https://github.com/pagopa/io-app-design-system/pull/393) in the `io-app-design-system` package.

I added a custom background color for the main screen, but it's probably the same for the other IDPay cards. We should refactor the cards to pass a proper color to both cards and `useHeaderSecondLevel`. What do you think?

**Comment by LeleDallas** [comment]
This commit be3271079e54e29739f20fc0725d4e6fe37cb5ec  solve some card header issue:

https://github.com/user-attachments/assets/36bc59db-162a-4206-ace2-ba84943b0f2d

@dmnplb let me know if you agree with this

**Comment by dmnplb** [comment]
> @dmnplb let me know if you agree with this

@LeleDallas LGTM :)

**Comment by dmnplb** [comment]
@LeleDallas I also made some changes to the other screens. In particular, I removed the `StatusBar` from the legacy `AppHeader` screen and the `RootContainer` main screen. Let's test these changes next week 💪🏻

**Comment by LeleDallas** [comment]
> @LeleDallas I also made some changes to the other screens. In particular, I removed the `StatusBar` from the legacy `AppHeader` screen and the `RootContainer` main screen. Let's test these changes next week 💪🏻

That's look good!

I just found another edge case in CGN details. 
👇 Check the video👇

https://github.com/user-attachments/assets/ce58a875-3109-4cb9-830f-fb6155f146f4

I believe in these cases, we need to replicate the same header color in the status bar (and also fix the action button colors)

**Comment by dmnplb** [comment]
@LeleDallas If we are sure that the regressions on light mode are pretty limited, what do you think about merging this PR and dealing with these edge cases later?

**Comment by LeleDallas** [comment]
> @LeleDallas If we are sure that the regressions on light mode are pretty limited, what do you think about merging this PR and dealing with these edge cases later?

I'll review each section again. However, since this change affects the entire app, I believe other streams should also review their sections

**Comment by LeleDallas** [comment]
> > @LeleDallas If we are sure that the regressions on light mode are pretty limited, what do you think about merging this PR and dealing with these edge cases later?
> 
> I'll review each section again. However, since this change affects the entire app, I believe other streams should also review their sections

I think for B&P stream we have to solve those issue here in this development:
| CGN | IDPay |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/953ce0a0-25cc-490f-8e5e-b9844efdcd2b"/> | <video src="https://github.com/user-attachments/assets/9ed00801-dc4e-4de9-9a0a-04051bfe363a" /> | 





For the status bar I suggest you to check this [document](https://reactnavigation.org/docs/status-bar/#tabs-and-drawer)

**Comment by dmnplb** [comment]
@LeleDallas Yesterday I replaced all the instances of `StatusBar` with the `FocusAwareStatusBar` ones (except for `IdentificationModal`). Just check all the screens of B&P flows and let me know if everything is okay 💪🏻

**Comment by LeleDallas** [comment]
> @LeleDallas Yesterday I replaced all the instances of `StatusBar` with the `FocusAwareStatusBar` ones (except for `IdentificationModal`). Just check all the screens of B&P flows and let me know if everything is okay 💪🏻

Good job! CGN and IDPay now looks great!🐉 

| CGN | IDPay |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/36ab47ad-8104-431e-b5a7-9ed49bd7d5a0"/> | <video src="https://github.com/user-attachments/assets/beaab740-60c9-44b7-93d6-d35b4887e105" /> | 

Maybe the only edge case I see are those:
- Android no status bar color
- iOS text unreadable 
Screenshot 2025-02-18 at 09 59 22

**Comment by dmnplb** [comment]
@LeleDallas Just fixed in the following commit:
* [Fix visual appearance of some CGN screens when dark mode is enabled · `8daeecd`](https://github.com/pagopa/io-app/pull/6680/commits/8daeecde3fea925547594be15831faf4ad28fda2)

**Comment by dmnplb** [comment]
@mastro993 Just added a custom `StatusBar` for `ItwCredential…` detail screen in the following commit:
* [Add support for custom StatusBar to ItwCredential… screen](https://github.com/pagopa/io-app/pull/6680/commits/4ff855bf6bf353b0dbca55b663f4ae525d46e1ad)

If you have time, take a look and see if everything looks okay.

**Review by LeleDallas — APPROVED** [review]
> @mastro993 Just added a custom `StatusBar` for `ItwCredential…` detail screen in the following commit:
> 
> * [Add support for custom StatusBar to ItwCredential… screen](https://github.com/pagopa/io-app/pull/6680/commits/4ff855bf6bf353b0dbca55b663f4ae525d46e1ad)
> 
> If you have time, take a look and see if everything looks okay.

Great work! B&P looks awesome now! 🚀

https://github.com/user-attachments/assets/72c89b62-7f3f-4488-8c0f-e96ff6f0e6cc
